File: floridopag/Lecture3/Lecture3examples/code/Java/readgames.java
/*
 * readgames.java
 * 
 * Copyleft 2016 Florido Paganelli <florido.paganelli@hep.lu.se>
 * 
 */
 
// import basic input/output java libraries 
import java.io.*;
// import java utility Scanner
import java.util.Scanner;

// everything is a class in java
public class readgames {
    // cause specific file errors in case of problems	
	public static void main (String args[]) throws FileNotFoundException, IOException {

        String text = new Scanner( new File("../../data/nintendowiigames.xml") ).useDelimiter("\\A").next();
        // try this code
        try {    
            // create an output buffer to standard output
            BufferedWriter output = new BufferedWriter(new OutputStreamWriter(System.out));
            // write the content of text on output
            output.write(text);
            // empty the content of standard out to screen
            output.flush();
        }
        // print an error if it fails
        catch (Exception e) {
            e.printStackTrace();
        }
	}
}

File: floridopag/Lecture3/Lecture3examples/code/CPP/readgames.cpp
/*
 * readgames.cpp
 * 
 * Copyleft 2016 Florido Paganelli <florido.paganelli@hep.lu.se>
 * 
 */

// library for basic input/output
#include <iostream>
// library for files stream
#include <fstream>
// library for strings stream
#include <sstream>
// library for strings
#include <string>
// if not specified, the functions belong to the std namespace
using namespace std;

int main(int argc, char **argv)
{
    // create a stream of strings
    std::stringstream filecontents;
    // create an input file stream
	ifstream myfile;
    // open the nintendowiigames.xml file as a file stream 
	myfile.open ("../../data/nintendowiigames.xml");
    // if the open was successfull
	if (myfile.is_open()) 
	{
       // stream the contents of the file inside the string stream
       filecontents << myfile.rdbuf();
	}
    // close the file
	myfile.close();
    // convert the stream to a string
    string contents(filecontents.str());
    // print out the string
    cout << contents;
	return 0;
}
